tests/Unit/Training/WaitingList/WaitingListTest.php

```diff
@@ -173,4 +177,37 @@ public function itCanHaveFlagsRemoved()
 
         $this->assertFalse($this->waitingList->flags->contains($flag));
     }
+
+    /** @test */
+    public function itSendsTopTenNotificationsOncePerAccount()
+    {
+        $status = WaitingListStatus::find(WaitingListStatus::DEFAULT_STATUS);
+        $accounts = factory(Account::class, 11)->create()->each(function ($account) use ($status) {
+            $atcSession = factory(Atc::class)->create(['account_id' => $account->id, 'minutes_online' => 721, 'disconnected_at' => now()]);
+            $this->waitingList->addToWaitingList($account, $this->privacc);
+            $this->waitingList->accounts()->each(function($waitingListAccount) use ($status) {
+                $waitingListAccount->pivot->addStatus($status);
+            });
+        });
+
+        Notification::fake();
+
+        $this->artisan('waitinglists:sendatctoptennotification');
+
+        Notification::assertTimesSent(10, WaitingListAtcTopTen::class);
+
+        Notification::fake();
```

Comment:
You should only need to call this once per test.

Reply:
Hi @CalumTowers 

From my testing, calling this resets the count of notifications, so without it being recalled, `assertNotSent` fails because the previous sub-test called the notification 10 times.

There are three options:

- Keep it as it is
- Do not reset the counts, and instead of asserting that a notification is not sent, assert that it hasn't changes (so still times sent = 10 from the previous sub-test)
- Split the test into 3 tests

In my opinion, one test is sufficient and there is no benefit to splitting it into 3 tests. I also believe it is cleaner / more readable the way that it currently is. I am asserting that a notification is not sent and it's clear why, where as it wouldn't be as clear if I was asserting that it was still sent 10 times.

Let me know your thoughts.

Reply:
I personally think what's here is fine. I did a bit of digging to see if there was a way to reset the counter that doesn't look like re-faking and couldn't easily find one.